src/background.js

```diff
@@ -18,6 +19,49 @@ import store from "./media-manager/store";
 
 const isDevelopment = process.env.NODE_ENV !== "production";
 
+// We need to ask macOS permission to access media devices
+async function getMediaPermission() {
+  let accessGranted = false;
+
+  accessGranted = await systemPreferences.askForMediaAccess("microphone");
+  accessGranted = await systemPreferences.askForMediaAccess("camera");
+
+  return accessGranted;
+}
+
+async function checkMediaPermission() {
+  const { platform } = process;
+
+  let macOSMediaDialogsAccepted = false;
+  let hasMediaPermission = false;
+
+  const microphoneAccessStatus = systemPreferences.getMediaAccessStatus(
+    "microphone"
+  );
+  const cameraAccessStatus = systemPreferences.getMediaAccessStatus("camera");
+
+  hasMediaPermission = microphoneAccessStatus === "granted";
+  hasMediaPermission = cameraAccessStatus === "granted";
+
+  if (platform === "darwin" && !hasMediaPermission) {
+    macOSMediaDialogsAccepted = await getMediaPermission();
+  } else if (platform === "darwin" && hasMediaPermission) {
+    macOSMediaDialogsAccepted = true;
+  }
+
+  if (
+    (platform === "win32" && !hasMediaPermission) ||
+    (platform === "darwin" && !macOSMediaDialogsAccepted)
+  ) {
+    dialog.showMessageBox({
+      type: "warning",
+      message: "modV does not have access to camera or microphone",
+      detail:
+        "While modV can still be used without these permissions, some functionality will be limited or broken. Please close modV, update your Security permissions and start modV again."
```

Comment:
In here I would refer to a specific help guide in the modV documentation (which doesn't exist yet) with steps someone might do if this is happening. And it would also be nice if we would know if it's the camera or the microphone, so the debugging can happen more specific.

Reply:
Can we create a new PR when we have the documentation, or do we want to wait?

Reply:
We can do this in a new PR.